pkg/apis/kubeflow/v1alpha2/types.go

```diff
@@ -0,0 +1,111 @@
+// Copyright 2019 The Kubeflow Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package v1alpha2
+
+import (
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// +genclient
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+type MPIJob struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+	Spec              MPIJobSpec `json:"spec,omitempty"`
+	Status            JobStatus  `json:"status,omitempty"`
+}
+
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+type MPIJobList struct {
+	metav1.TypeMeta `json:",inline"`
+	metav1.ListMeta `json:"metadata"`
+	Items           []MPIJob `json:"items"`
+}
+
+type ProcessingSpec struct {
+	// Specifies the desired number of processing units the MPIJob should run on.
+	// Mutually exclusive with `ReplicaSpec.Replicas` in `MPIReplicaSpecs`.
+	// +optional
+	Units *int32 `json:"units,omitempty"`
```

Comment:
Will these be converted to replica pod spec.. what about memory in the case of cpu type?

Reply:
Good question. This is not currently considered inside `ProcessingSpec`. We could add memory, disk, etc. to the spec. Another idea is that we could get rid of `ProcessingSpec` and make fully use of `MPIReplicaSpecs` which will allow users to provide specific pod spec. One feedback I often receive from our users is that having both `ProcessingSpec` and `MPIReplicaSpecs` are confusing. @rongou What do you think the best approach would be?

Reply:
I'm fine with getting rid of `ProcessingSpec`.

Reply:
Sounds good. @madhukarkm @rongou I've just removed ProcessingSpec. Please take another look when you get a chance. This is just initial API spec. I also added some TODOs that will be addressed later once we have a common operator.